consul monitor log-json do not produce json format logs

When filing a bug, please include the following headings if possible. Any example text in this template can be deleted.

#### Overview of the Issue
When starting `consul monitor  -log-json` the logs are not in json format

#### Reproduction Steps
- `consul agent` in tab1
- `consul monitor  -log-json`  in another tab
- notice the logs are not in json

### Operating system and Environment details

The bug is os independant

### Log Fragments

```
consul monitor -log-level=DEBUG  -log-json
2021-06-07T10:43:01.061-0400 [INFO]  agent: Caught: signal=interrupt
2021-06-07T10:43:01.061-0400 [INFO]  agent: Graceful shutdown disabled. Exiting
2021-06-07T10:43:01.061-0400 [INFO]  agent: Requesting shutdown
2021-06-07T10:43:01.061-0400 [INFO]  agent.server: shutting down server
2021-06-07T10:43:01.061-0400 [DEBUG] agent.leader: stopping routine: routine="federation state anti-entropy"
2021-06-07T10:43:01.061-0400 [DEBUG] agent.leader: stopping routine: routine="federation state pruning"
2021-06-07T10:43:01.061-0400 [DEBUG] agent.leader: stopping routine: routine="intermediate cert renew watch"
2021-06-07T10:43:01.061-0400 [DEBUG] agent.server.usage_metrics: usage metrics reporter shutting down
2021-06-07T10:43:01.061-0400 [DEBUG] agent.leader: stopped routine: routine="federation state pruning"
2021-06-07T10:43:01.061-0400 [DEBUG] agent.leader: stopping routine: routine="CA root pruning"
2021-06-07T10:43:01.062-0400 [DEBUG] agent.leader: stopped routine: routine="intermediate cert renew watch"
2021-06-07T10:43:01.062-0400 [WARN]  agent.server.serf.lan: serf: Shutdown without a Leave
2021-06-07T10:43:01.062-0400 [DEBUG] agent.leader: stopped routine: routine="CA root pruning"
2021-06-07T10:43:01.062-0400 [ERROR] agent.server: error performing anti-entropy sync of federation state: error="context canceled"
2021-06-07T10:43:01.062-0400 [DEBUG] agent.leader: stopped routine: routine="federation state anti-entropy"
2021-06-07T10:43:01.062-0400 [DEBUG] agent.server.autopilot: state update routine is now stopped
2021-06-07T10:43:01.062-0400 [DEBUG] agent.server.autopilot: autopilot is now stopped
2021-06-07T10:43:01.062-0400 [WARN]  agent.server.serf.wan: serf: Shutdown without a Leave
2021-06-07T10:43:01.062-0400 [INFO]  agent.router.manager: shutting down
2021-06-07T10:43:01.063-0400 [INFO]  agent.router.manager: shutting down
2021-06-07T10:43:01.063-0400 [INFO]  agent: consul server down
2021-06-07T10:43:01.063-0400 [INFO]  agent: shutdown complete
2021-06-07T10:43:01.063-0400 [INFO]  agent: Stopping server: protocol=DNS address=127.0.0.1:8600 network=tcp
2021-06-07T10:43:01.063-0400 [INFO]  agent: Stopping server: protocol=DNS address=127.0.0.1:8600 network=udp
2021-06-07T10:43:01.064-0400 [INFO]  agent: Stopping server: address=127.0.0.1:8500 network=tcp protocol=http
2021-06-07T10:43:02.069-0400 [WARN]  agent: Failed to stop server: address=127.0.0.1:8500 network=tcp protocol=http
2021-06-07T10:43:02.069-0400 [INFO]  agent: Waiting for endpoints to shut down
2021-06-07T10:43:02.069-0400 [INFO]  agent: Endpoints down
```

--- a/.changelog/10358.txt
+++ b/.changelog/10358.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+monitor: fix monitor to produce json format logs when requested
+```

--- a/command/monitor/monitor.go
+++ b/command/monitor/monitor.go
@@ -68,11 +68,12 @@ func (c *cmd) Run(args []string) int {
 			c.UI.Error(fmt.Sprintf("Error starting JSON monitor: %s", err))
 			return 1
 		}
-	}
-	logCh, err = client.Agent().Monitor(c.logLevel, eventDoneCh, nil)
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error starting monitor: %s", err))
-		return 1
+	} else {
+		logCh, err = client.Agent().Monitor(c.logLevel, eventDoneCh, nil)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error starting monitor: %s", err))
+			return 1
+		}
 	}
 
 	go func() {

--- a/command/monitor/monitor_test.go
+++ b/command/monitor/monitor_test.go
@@ -1,6 +1,8 @@
 package monitor
 
 import (
+	"encoding/json"
+	"strings"
 	"sync"
 	"testing"
 	"time"
@@ -117,6 +119,69 @@ func TestMonitorCommand_LogJSONValidFlag(t *testing.T) {
 		}
 		// OK!
 	case <-time.After(1 * time.Second):
+		t.Fatal("timed out waiting for exit")
+	}
+}
+
+func TestMonitorCommand_LogJSONValidFormat(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+	a := agent.StartTestAgent(t, agent.TestAgent{})
+	defer a.Shutdown()
+
+	shutdownCh := make(chan struct{})
+
+	ui := cli.NewMockUi()
+	c := New(ui, shutdownCh)
+	args := []string{"-http-addr=" + a.HTTPAddr(), "-log-json"}
+
+	// Buffer it so we don't deadlock when blocking send on shutdownCh triggers
+	// Run to return before we can select on it.
+	exitCode := make(chan int, 1)
+
+	// Run the monitor in another go routine. If this doesn't exit on our "signal"
+	// then the whole test will hang and we'll panic (to not blow up if people run
+	// the suite without -timeout)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		wg.Done() // Signal that this goroutine is at least running now
+		exitCode <- c.Run(args)
+	}()
+
+	// Wait for that routine to at least be running
+	wg.Wait()
+
+	// Read the logs and try to json marshall it
+	go func() {
+		time.Sleep(1 * time.Second)
+		outputs := ui.OutputWriter.String()
+		for count, output := range strings.Split(outputs, "\n") {
+			if output != "" && count > 0 {
+				jsonLog := new(map[string]interface{})
+				err := json.Unmarshal([]byte(output), jsonLog)
+				if err != nil {
+					exitCode <- -1
+				}
+				if len(*jsonLog) <= 0 {
+					exitCode <- 1
+				}
+			}
+		}
+		shutdownCh <- struct{}{}
+
+	}()
+
+	select {
+	case ret := <-exitCode:
+		if ret != 0 {
+			t.Fatal("command returned with non-zero code")
+		}
+		// OK!
+	case <-time.After(5 * time.Second):
 		t.Fatal("timed out waiting for exit")
 	}
 }

